federation/pkg/federation-controller/util/federated_informer.go

```diff
@@ -146,9 +164,16 @@ func NewFederatedInformer(federationClient federation_release_1_4.Interface, tar
 					return
 				}
 				if isClusterReady(oldCluster) != isClusterReady(curCluster) || !reflect.DeepEqual(oldCluster.Spec, curCluster.Spec) {
+
+					if clusterLifecycle.ClusterUnavailable != nil {
+						clusterLifecycle.ClusterUnavailable(oldCluster, federatedInformer, federatedInformer.GetTargetStore())
+					}
 					federatedInformer.deleteCluster(oldCluster)
 					if isClusterReady(curCluster) {
 						federatedInformer.addCluster(curCluster)
+						if clusterLifecycle.ClusterAvailable != nil {
+							clusterLifecycle.ClusterAvailable(curCluster, federatedInformer, federatedInformer.GetTargetStore())
+						}
```

Comment:
In spec there is only an ip address of a cluster. If the cluster changes ip there can be a good chance that the content of the cluster is different. I think that cluster ip address change is so rare that we don't need to be super efficient here.